lib/uart.cpp

```diff
@@ -98,47 +122,59 @@ uint8_t Uart::read(Uart::Address register_address) {
                 }
                 return data;
             }
-            break;
         }
         case Uart::Address::IER: {//IER or DLM
             if (this->dlab()) {//DLM
                 return this->regs.m_dlm;
             } else {//IER
-                assert(false && "TODO");//TODO
+                return this->regs.m_ier;
             }
-            break;
         }
         case Uart::Address::ISR: {//NOTE: Never FCR since that isn't readable
-            assert(false && "TODO");//TODO
-            break;
+            uint8_t isr = 0;
+            isr |= (this->regs.m_fcr & 0b1) ? (0b11 << 6) : 0;//Indicate the FIFOs are enabled if they infact are
```

Comment:
Using the ternary operator here is clever but using `if` would make it clearer what's being done. Also, why 0b11? There are a bunch of magic numbers here, might be worthwhile to make some constants like you did for LSR.

Reply:
Yeah it was a bit of a rush job since I was excited :)

I'll clean things up.